File: 3546_Equal_Sum_Grid_Partition_I/solution.c
bool canPartitionGrid(int** grid, int gridSize, int* gridColSize) {
    int m = gridSize, n = gridColSize[0];
    long long t = 0;
    for (int i = 0; i < m; i++) for (int j = 0; j < n; j++) t += grid[i][j];
    if (t % 2) return 0;
    long long a = t / 2;

    long long s = 0;
    for (int i = 0; i < m - 1; i++) {
        for (int j = 0; j < n; j++) s += grid[i][j]; if (s == a) return 1;
    }
    s = 0;
    for (int j = 0; j < n - 1; j++) {
        for (int i = 0; i < m; i++) s += grid[i][j]; if (s == a) return 1;
    }
    return 0;
}

File: 3546_Equal_Sum_Grid_Partition_I/solution.py
class Solution:
    def canPartitionGrid(self, grid: List[List[int]]) -> bool:
        m, n = len(grid), len(grid[0])
        t = sum(sum(r) for r in grid)

        if t % 2: return False
        a = t // 2

        s = 0
        for i in range(m - 1):
            s += sum(grid[i])
            if s == a: return True
        s = 0
        for j in range(n - 1):
            for i in range(m): s += grid[i][j]
            if s == a: return True
            
        return False
